webapp/src/Service/SubmissionService.php

```diff
@@ -453,9 +453,10 @@ public function submitSolution(
             throw new BadRequestHttpException('Submissions for contest (temporarily) disabled');
         }
 
-        if (!$language->getAllowSubmit()) {
+        $allowedLanguages = $this->dj->getAllowedLanguagesForContest($contest);
+        if (!in_array($language, $allowedLanguages, true)) {
```

Comment:
Nitpick, maybe consider:
```suggestion
        if (!in_array($language, $allowedLanguages, strict: true)) {
```